### ISSUE TITLE
Pods won't spin up in rke2 cluster - Stuck in Error from Server (Bad Request): Container Creating 

### ISSUE THREAD

**Issue by kurborg** [rancher/rke2#5700]
So I have fully stood up all my nodes in my rke2 cluster. I am trying to spin up traefik and another testhelloworld application but the pods are stuck in 
`Error from server (BadRequest): container "traefik" in pod "traefik-deployment-7b66896dcb-zk742" is waiting to start: ContainerCreating`
`Error from server (BadRequest): container "testhelloworld" in pod "testhelloworld-5496fdfb46-jhcfx" is waiting to start: ContainerCreating`

I have harbor up that my k8 resources are pulling images from, I don't think that's the issue as it's not complaining about "Backoff Pull Image Request."
I do have firewalld back up since it's required to run docker-compose to deploy harbor, but the rke2 nodes all seem to be fine if I spin up all the nodes then reactivate firewalld. I don't see how that would stop pods from creating containers but could that be an issue? 

![image](https://github.com/rancher/rke2/assets/45808976/a6100d57-16ca-47f4-b6b0-ab0d4dea7229)

**Comment by kurborg** [comment]
Running 
`kubectl describe` on the pod it's giving me: Pod sandbox changed

Events:
  Type    Reason          Age                    From     Message
  ----    ------          ----                   ----     -------
  Normal  SandboxChanged  2m12s (x2371 over 8h)  kubelet  Pod sandbox changed, it will be killed and re-created.

**Comment by kurborg** [comment]
I also see that I have a core-dns pod not spinning up.. the other is working fine it seems 

![image](https://github.com/rancher/rke2/assets/45808976/c4bffea6-db5a-4f0d-a507-8f780ec05327)

kubectl logs gives 

![image](https://github.com/rancher/rke2/assets/45808976/841b5b4b-2a73-4c6f-8092-0ee2976aa4a4)

**Comment by brandond** [comment]
You're running firewalld and docker alongside rke2 on your nodes? And you're stopping/restarting firewalld because you're managing some stuff using docker-componse instead of running it in Kubernetes, and it only works when firewalld is running? This is not something we test; I would probably recommend not running your docker-compose workloads on the same host as RKE2, and removing docker and firewalld.

**Comment by kurborg** [comment]
Okay Brandon, I've disabled firewalld and docker but still getting this error when running kubectl apply on my deployment

`
  Warning  Failed     39s (x6 over 113s)  kubelet            Error: ImagePullBackOff
  Normal   Pulling    26s (x4 over 113s)  kubelet            Pulling image "traefik:v2.11"
  Warning  Failed     26s (x4 over 113s)  kubelet            Failed to pull image "traefik:v2.11": rpc error: code = Unknown desc = failed to pull and unpack image "traefik:v2.11": failed to resolve reference "traefik:v2.11": failed to do request: Head "traefik/manifests/v2.11": tls: failed to verify certificate: x509: certificate signed by unknown authority
`

I've added a secret docker-registry that is valid to my cluster and my KUBECONFIG points to my rke2.yaml that contains the right certificates. I'm able to pull the images locally just fine

**Comment by kurborg** [comment]
I've attempted using kubectl apply with --insecure-skip-tls-verify also and still same error

**Comment by kurborg** [comment]
For reference, I'm using a private image repository that is signed by a custom CA. I have the ca.crt, do I need to copy it somewhere for RKE2 to trust my CA?

**Comment by brandond** [comment]
Have you configured registry mirrors in registries.yaml, or are you just going through a traditional HTTP proxy to reach Docker Hub?

**Comment by kurborg** [comment]
@brandond So that's pulling from my private harbor registry. This is an airgapped environment so I can't use docker hub. I have a private registry at a private url that I removed from that error message but it basically isn't trusting it. I know I had to add my ca.crt to my /etc/docker/certs.d/<URL>/ in order to be able to pull images, is there a similar thing I need to do for my rke2 cluster? I did create a docker-registry secret within the cluster and added each container spec to have a imagePullSecrets for that, but still not trusting it....
